src/node_i18n.cc

```diff
@@ -502,18 +503,32 @@ void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
       }
     }
     ret = ToBufferEndian(env, &result);
```

Comment:
Do we even need `ret` as a temporary `Buffer` instance here? It seems like we could just use `result` throughout the rest of the function instead.

Reply:
I added a new commit with the changes recommended, and here's the updated benchmark:

```
                                                                            confidence improvement accuracy (*)    (**)   (***)
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='iso-8859-3'         ***     19.20 %       ±8.41% ±11.19% ±14.58%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='latin1'             ***     29.28 %       ±9.39% ±12.52% ±16.37%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='utf-8'                *      9.80 %       ±8.89% ±11.84% ±15.41%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='iso-8859-3'          **     12.22 %       ±8.09% ±10.77% ±14.04%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='latin1'             ***     20.98 %       ±8.68% ±11.55% ±15.04%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='utf-8'                *      9.26 %       ±9.07% ±12.08% ±15.75%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='iso-8859-3'           ***     33.18 %       ±4.59%  ±6.11%  ±7.96%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='latin1'               ***     37.25 %       ±7.19%  ±9.58% ±12.49%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='utf-8'                ***     27.66 %       ±6.38%  ±8.53% ±11.17%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='iso-8859-3'           ***     30.96 %       ±7.13%  ±9.50% ±12.38%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='latin1'               ***     32.87 %       ±6.23%  ±8.29% ±10.79%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='utf-8'                ***     29.71 %       ±6.21%  ±8.26% ±10.75%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='iso-8859-3'          *      4.03 %       ±3.31%  ±4.40%  ±5.73%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='latin1'                    -1.20 %       ±4.34%  ±5.79%  ±7.58%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='utf-8'                      0.89 %      ±10.20% ±13.57% ±17.66%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='iso-8859-3'                 4.78 %       ±5.46%  ±7.28%  ±9.50%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='latin1'                    -0.30 %       ±3.72%  ±4.95%  ±6.44%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='utf-8'                     -4.20 %      ±10.79% ±14.37% ±18.75%

Be aware that when doing many comparisons the risk of a false-positive result increases.
In this case, there are 18 comparisons, you can thus expect the following amount of false-positive results:
  0.90 false positives, when considering a   5% risk acceptance (*, **, ***),
  0.18 false positives, when considering a   1% risk acceptance (**, ***),
  0.02 false positives, when considering a 0.1% risk acceptance (***)
```